core/lib/Tab.ts

```diff
@@ -258,6 +258,71 @@ export default class Tab
     interceptor.urls = blockedUrls;
   }
 
+  public setInterceptedResources(interceptedResources?: InterceptedResource[]): void {
+    const mitmSession = this.session.mitmRequestSession;
+
+    let interceptor = mitmSession.interceptorHandlers.find(x => x.types && !x.handlerFn);
```

Comment:
I think this is going to end up conflicting with other possible uses of the interceptor handlers. The code you copied is putting in a "default" handler, but I don't think that's what you want here. You'd messing with interceptors that are put in by other users/parts of the system. I think you want to loop through your interceptedResource list and add an interceptor for each url that only handles that specific url. You could check (or not) whether an existing handler exists. I think since you've only set this up to work via the constructor, you can skip the check.